src/commands/new.ts

```diff
@@ -12,84 +12,291 @@ import {
     const { exists, path, remove, copy, read, write } = filesystem
     const { info, colors, warning } = print
     const { gray, red, magenta, cyan, yellow, green } = colors
+    const options: Options = parameters.options
+
+    const y = !!options.y
+    const useDefault = (option: unknown) => y && option === undefined
+    // #endregion
 
+    // #region Debug
     // start tracking performance
     const perfStart = new Date().getTime()
 
     // debug?
-    const debug = Boolean(parameters.options.debug)
+    const debug = Boolean(options.debug)
     const log = <T = unknown>(m: T): T => {
       debug && info(` ${m}`)
       return m
     }
 
     // log raw parameters for debugging
     log(`ignite command: ${parameters.argv.join(" ")}`)
+    // #endregion
+
+    // #region Project Name
+    heading("🔥 About to Ignite your new app! 🔥")
 
     // retrieve project name from toolbox
-    const { validateProjectName, validateBundleIdentifier } = require("../tools/validations")
-    const projectName = validateProjectName(toolbox)
+    const { validateProjectName } = require("../tools/validations") as ValidationsExports
+    const projectName = await validateProjectName(toolbox)
     const projectNameKebab = kebabCase(projectName)
+    // #endregion
 
-    // if they pass in --overwrite, remove existing directory otherwise throw if exists
-    if (parameters.options.overwrite) {
-      log(`Removing existing project ${projectName}`)
-      remove(projectName)
-    } else if (exists(projectName)) {
-      const alreadyExists = `Error: There's already a folder with the name "${projectName}". To force overwriting that folder, run with --overwrite.`
-      p()
-      p(yellow(alreadyExists))
-      process.exit(1)
-    }
-
+    // #region Boilerplate
     // if they pass in --boilerplate, warn them to use old Ignite
-    const bname = parameters.options.b || parameters.options.boilerplate
+    const bname = options.b || options.boilerplate
     if (bname) {
       p()
       p(yellow(`Different boilerplates are no longer supported in Ignite v4+.`))
       p(gray(`To use the old CLI to support different boilerplates, try:`))
       p(cyan(`npx ignite-cli@3 new ${projectName} --boilerplate ${bname}`))
       process.exit(1)
     }
+    // #endregion
+
+    // #region Bundle Identifier
+    const defaultBundleIdentifier = `com.${projectName.toLowerCase()}`
+    let bundleIdentifier = useDefault(options.bundle) ? defaultBundleIdentifier : options.bundle
+
+    if (bundleIdentifier === undefined) {
+      const bundleIdentifierResponse = await prompt.ask({
+        type: "input",
+        name: "bundleIdentifier",
+        message: "What bundle identifier?",
+        initial: defaultBundleIdentifier,
+      })
+
+      bundleIdentifier = bundleIdentifierResponse.bundleIdentifier
+    }
+
+    const { validateBundleIdentifier } = require("../tools/validations") as ValidationsExports
+    validateBundleIdentifier(toolbox, bundleIdentifier)
```

Comment:
Usage of this has slightly changed. `validateBundleIdentifier` returns a boolean, but we only care about calling it will log a warning and call `process.exit(1)` if there is an error.